### PR TITLE
Update ch08

### PR DESCRIPTION
**Pull request by waddlaw**
Changes:

- Fix example
- Fix minor typos
- Change type `List a` to `[a]`
- Rearrange base case position
- Add type signature, so that fix `LH` error
- Fix the missing base case in `insert` function

### CODE REVIEW

File: src/08-measure-sets.lhs [hunk 1]
```diff
@@ -395,6 +395,7 @@ the appropriate position of the output:
 
 \begin{code}
 insert x []     = [x]
+insert x [x]    = [x]
```
Comment (ranjitjhala) [inline src/08-measure-sets.lhs]: Hmm. This is not valid Haskell code, right? (Because `x` is bound twice?)

File: src/08-measure-sets.lhs [hunk 2]
```diff
@@ -439,8 +440,8 @@ prop_merge_app xs ys = elts zs == elts zs'
 
 <div class="hwex" id="Merge Sort">
 \doublestar Once you write the correct type
-for `merge` above, you should be able to prove the
-er, unexpected signature for `mergeSort` below.
+for `merge` above, you should be able to prove thet
```
Comment (ranjitjhala) [inline src/08-measure-sets.lhs]: Should this typo be: `thet` --> `the` ?

**Comment by ranjitjhala** [comment]
This is wonderful, thanks @waddlaw !!! I made two comments, one is a simple typo. The other is in the `insert x [x]` should probably not be there right? (Also I think it would remove duplicates from the list which is perhaps undesirable?)

File: src/08-measure-sets.lhs (hunk 2 above)
Comment (waddlaw) [reply src/08-measure-sets.lhs]: Sorry, You're right!

File: src/08-measure-sets.lhs (hunk 1 above)
Comment (waddlaw) [reply src/08-measure-sets.lhs]: Sorry!! I mistook it as `mergeSort`.I will correct it soon.

**Comment by waddlaw** [comment]
Thanks of review! Fixed the two point.

If lack of the base case of `mergeSort` then no-termination.

**Comment by ranjitjhala** [comment]
Fantastic, thanks!!! 👍

**Pull request merged**